planetscale/branches.go

```diff
@@ -89,12 +89,30 @@ type RefreshSchemaRequest struct {
 	Branch       string `json:"-"`
 }
 
-// PromoteBranchRequest encapsulates the request for promoting a branch to
+// PromoteRequest encapsulates the request for promoting a branch to
 // production.
-type PromoteBranchRequest struct {
+type PromoteRequest struct {
 	Organization string `json:"-"`
 	Database     string `json:"-"`
-	Branch       string `json:"branch"`
+	Branch       string `json:"-"`
+}
+
+type GetPromotionRequestRequest struct {
+	Organization string `json:"-"`
+	Database     string `json:"-"`
+	Branch       string `json:"-"`
+}
+
+// BranchPromotionRequest represents a promotion request for a branch.
+type BranchPromotionRequest struct {
+	ID         string     `json:"id"`
+	Branch     string     `json:"branch"`
+	LintErrors string     `json:"lint_errors"`
+	State      string     `json:"state"`
```

Comment:
I think we should provide enums instead of strings in this case. There are probably only a couple states it can be, and it's going to be helpful for the user of the SDK (and even for us) to know the possible values.

Reply:
@fatih How would you suggest doing the enums? Just using an `iota`? I feel like we had another example of this in the SDK...I can try and find it.

Reply:
Yeah checkout `ReviewaAction` here: https://github.com/planetscale/planetscale-go/blob/16cffb488f8a29c33d78ce1818b762e851f9aa6f/planetscale/deploy_requests.go#L157